Dining/views.py

```diff
@@ -478,51 +478,26 @@ def post(self, request, *args, **kwargs):
             return self.render_to_response(context)
 
 
-class IsOwnerMixin(object):
-    """
-    Ensures the page can only be visited by dining list owners
-
-    Attributes:
-        redirect_name: The url name to redirect to upon rejection
-        base_redirect: the actual url to redirect to, can be called when succesfully processing the page
-    """
-    redirect_name = "slot_details"
-
-    @property
-    def base_redirect(self):
-        return self.reverse(self.redirect_name)
+class SlotOwnerMixin(SlotMixin):
+    """Slot mixin extension that makes it only accessible for dining list owners."""
 
     def dispatch(self, request, *args, **kwargs):
-        """
-        Construct date before get/post is called.
-        """
+        # Initialise dining list
+        self.init_dining_list()
+        # Check permission
         if not self.dining_list.is_owner(request.user):
-            if hasattr(self, "base_redirect"):
-                messages.error(request, "You are not the owner of this list and can't do this action")
-                return HttpResponseRedirect(self.base_redirect)
-            else:
-                return HttpResponseForbidden()
+            raise PermissionDenied
+        # Dispatch
         return super().dispatch(request, *args, **kwargs)
 
 
-class SlotOwnerMixin(LoginRequiredMixin, SlotMixin, IsOwnerMixin):
-    """
-    Combined mixin for slot pages that should only be accessable for dining list owners
-    """
-    pass
-
```

Comment:
The reason I implemented it like this is otherwise, when calling the SlotownerMixin the LoginRequiredMixin must also be called in every child in order to prevent an error in running the code to confirm the logged in user is owner. Arguably this is also the case for SlotMixin as that creates visittrackers and could be moved there so could be better there.

Anyhow, the current removal does not address the lack of loginchecks, either revert the LoginRequired back to each subchild or implement it in some way in the parent. As the order of parent processing is important I propose something similar to what I did above, though I'd love to hear your opinion on that.